File: problems/Add Two Numbers /addTwoNumbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = null;
        ListNode node = null;
        var memory = 0;

        while (null != l1 || null != l2 || 1 == memory) {
            var curr = memory;
            if (null != l1) {
                curr += l1.val;
                l1 = l1.next;
            }
            if (null != l2) {
                curr += l2.val;
                l2 = l2.next;
            }
            if (null == head) {
                head = new ListNode(curr % 10);
                node = head;
            } else {
                node.next = new ListNode(curr % 10);
                node = node.next;
            }
            memory = curr / 10;
        }

        return head;
    }
}

File: problems/Add Two Numbers /addTwoNumbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode fakeHead = new ListNode();
        ListNode curr = fakeHead;
        int carry = 0;

        while (null != l1 || null != l2 || 0 != carry) {
            int sum = carry;

            if (null != l1) {
                sum += l1.val;
                l1 = l1.next;
            }

            if (null != l2) {
                sum += l2.val;
                l2 = l2.next;
            }

            curr.next = new ListNode(sum % 10);
            curr = curr.next;
            carry = sum / 10;
        }

        return fakeHead.next;
    }
}
